File: app/src/main/java/com/example/hyunju/icon_mobileproject/MainActivity.java
package com.example.hyunju.icon_mobileproject;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.concurrent.ExecutionException;

import static android.provider.AlarmClock.EXTRA_MESSAGE;

public class MainActivity extends AppCompatActivity {


    TextView tv ;
    ListView listview;
    ArrayList arr;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        listview = (ListView)findViewById(R.id.listView1);
        BlockAPITask t = new BlockAPITask();
        try {
           arr= t.execute().get();
            ArrayAdapter<String> itemsAdapter =
                    new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, arr);


            listview.setAdapter(itemsAdapter);
            listview.setOnItemClickListener(new AdapterView.OnItemClickListener() {
                @Override
                public void onItemClick(AdapterView<?> parent, View view, int position,
                                        long id) {
                    Intent intent = new Intent(MainActivity.this, Main2Activity.class);

                    intent.putExtra("block_hash", (String) arr.get(position));
                    startActivity(intent);
                }
            });
        } catch (InterruptedException e) {
            e.printStackTrace();
        } catch (ExecutionException e) {
            e.printStackTrace();
        }


    }
}

File: app/src/main/java/com/example/mobile_tracker/MainActivity.kt
package com.example.mobile_tracker

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val blocks = Request(ArrayList()).execute()

        val listAdapter = ListAdapter(this, ArrayList(blocks.get()))
        blockListView.adapter = listAdapter

        my_swipeRefresh_Layout.setOnRefreshListener {
            Request(blocks.get()).execute()
        }
    }
}
